Source/UIView+SwiftCarousel.swift

```diff
@@ -20,15 +20,24 @@
 * THE SOFTWARE.
 */
 
-extension UIView {
+public enum ArchiveCopyingError: ErrorType {
+    case View
+}
+
+public extension UIView {
+    private func prepareConstraintsForArchiving() {
+        constraints.forEach { $0.shouldBeArchived = true }
+        subviews.forEach { $0.prepareConstraintsForArchiving() }
+    }
+    
     /**
      Method to copy UIView using archivizing.
      
      - returns: Copied UIView (different memory address than current)
      */
-    func copyView() -> UIView {
-        let serialized = NSKeyedArchiver.archivedDataWithRootObject(self)
-        
-        return NSKeyedUnarchiver.unarchiveObjectWithData(serialized) as! UIView
+    public func copyView() throws -> UIView {
+        prepareConstraintsForArchiving()
+        guard let view = NSKeyedUnarchiver.unarchiveObjectWithData(NSKeyedArchiver.archivedDataWithRootObject(self)) as? UIView else { throw ArchiveCopyingError.View }
+        return view
```

Comment:
This is a good improvement, but in my use case the view has a reference to a model object, which it uses to layout it subviews depending on the state of the model. 

I changed my code so that I store references to my models in a separate array, and the reference to the corresponding model object is always set to the view on `didSelect` and `didDeselect` delegate methods, before calling the method that updates the layout in my view, and that did the trick for me. 

I'm thinking that maybe we can just write some notes in the SwiftCarousel docs about this `copyView` limitation as a warning for anybody running into this in the future, what do you think? In other projects I have more complex views that have a reference to a custom delegate and gesture recognisers, and I think those wouldn't be properly be copied by this implementation neither.